File: push/src/main/java/com/xiaomi/slim/PacketParser.java
package com.xiaomi.slim;

import com.xiaomi.smack.Connection;
import com.xiaomi.smack.XMPPException;
import com.xiaomi.smack.packet.Message;
import com.xiaomi.smack.packet.Packet;
import com.xiaomi.smack.util.PacketParserUtils;
import java.io.ByteArrayInputStream;
import java.io.InputStreamReader;
import org.xmlpull.v1.XmlPullParser;
import org.xmlpull.v1.XmlPullParserException;
import org.xmlpull.v1.XmlPullParserFactory;

/* JADX INFO: loaded from: miuipushsdkshared_3_7_9.jar:com/xiaomi/slim/PacketParser.class */
public class PacketParser {
    private XmlPullParser mParser;

    PacketParser() {
        try {
            XmlPullParser xmlPullParserNewPullParser = XmlPullParserFactory.newInstance().newPullParser();
            this.mParser = xmlPullParserNewPullParser;
            xmlPullParserNewPullParser.setFeature("http://xmlpull.org/v1/doc/features.html#process-namespaces", true);
        } catch (XmlPullParserException e) {
        }
    }

    Packet parse(byte[] bArr, Connection connection) throws Exception {
        this.mParser.setInput(new InputStreamReader(new ByteArrayInputStream(bArr)));
        this.mParser.next();
        int eventType = this.mParser.getEventType();
        String name = this.mParser.getName();
        if (eventType != 2) {
            return null;
        }
        if (name.equals("message")) {
            return PacketParserUtils.parseMessage(this.mParser);
        }
        if (name.equals("iq")) {
            return PacketParserUtils.parseIQ(this.mParser, connection);
        }
        if (name.equals("presence")) {
            return PacketParserUtils.parsePresence(this.mParser);
        }
        if (this.mParser.getName().equals("stream")) {
            return null;
        }
        if (this.mParser.getName().equals(Message.MSG_TYPE_ERROR)) {
            throw new XMPPException(PacketParserUtils.parseStreamError(this.mParser));
        }
        if (!this.mParser.getName().equals("warning")) {
            this.mParser.getName().equals("bind");
            return null;
        }
        this.mParser.next();
        this.mParser.getName().equals("multi-login");
        return null;
    }
}

File: legacy/src/main/java/com/xiaomi/slim/PacketParser.kt
package com.xiaomi.slim
import io.github.magisk317.mipush.protocol.model.*

import com.xiaomi.smack.Connection
import com.xiaomi.smack.XMPPException
import com.xiaomi.smack.packet.Message
import com.xiaomi.smack.packet.Packet
import com.xiaomi.smack.util.PacketParserUtils
import org.xmlpull.v1.XmlPullParser
import org.xmlpull.v1.XmlPullParserFactory
import java.io.ByteArrayInputStream
import java.io.InputStreamReader

class PacketParser {
    private val mParser: XmlPullParser

    init {
        val factory = XmlPullParserFactory.newInstance()
        mParser = factory.newPullParser()
        mParser.setFeature("http://xmlpull.org/v1/doc/features.html#process-namespaces", true)
    }

    @Throws(Exception::class)
    fun parse(
        data: ByteArray,
        connection: Connection,
    ): Packet? {
        mParser.setInput(InputStreamReader(ByteArrayInputStream(data)))
        mParser.next()
        val eventType = mParser.eventType
        val name = mParser.name

        if (eventType != XmlPullParser.START_TAG) {
            return null
        }

        return when (name) {
            "message" -> PacketParserUtils.parseMessage(mParser)
            "iq" -> PacketParserUtils.parseIQ(mParser, connection)
            "presence" -> PacketParserUtils.parsePresence(mParser)
            "stream" -> null
            Message.MSG_TYPE_ERROR -> throw XMPPException(PacketParserUtils.parseStreamError(mParser))
            "warning" -> {
                mParser.next()
                mParser.name == "multi-login"
                null
            }
            "bind" -> null
            else -> null
        }
    }
}
